storage/src/mmr/bitmap.rs

```diff
@@ -428,20 +421,30 @@ impl<H: CHasher, const N: usize> Bitmap<H, N> {
         let leaf_pos = Self::leaf_pos(bit_offset);
         let chunk = self.get_chunk(bit_offset);
 
+        if leaf_pos == self.mmr.size() {
+            assert!(self.next_bit > 0);
+            // Proof is over a bit in the partial chunk. In this case only a single digest is
+            // required in the proof: the mmr's root.
+            return Ok((
+                Proof {
+                    size: self.bit_count(),
+                    digests: vec![self.mmr.root(hasher)],
+                },
+                *chunk,
+            ));
+        }
+
+        let mut proof = Proof::<H>::range_proof(&self.mmr, leaf_pos, leaf_pos).await?;
+        proof.size = self.bit_count();
         if self.next_bit == 0 {
-            let proof = Proof::<H>::range_proof(&self.mmr, leaf_pos, leaf_pos).await?;
+            // Bitmap is chunk aligned.
```

Comment:
nit: I think it is clearer to put this just above 437

Reply:
Maybe I'm misunderstanding, are you suggesting moving the comment there?  (At L437 we don't know if the bitmap is chunk aligned...)

Reply:
Did you mean before L439?

Reply:
I was just thinking to put above the `let mut proof` line (I sometimes do that if the intermediate lines between the comment are self-explanatory and another `\n` doesn't look right). Your call lol.

Reply:
Hmm, I think that's going to be confusing since we use that proof even if it's not chunk-aligned (we just mutate it a bit more).